src/node_report_utils.cc

```diff
@@ -82,6 +82,38 @@ static void ReportEndpoints(uv_handle_t* h, JSONWriter* writer) {
   ReportEndpoint(h, rc == 0 ? addr : nullptr, "remoteEndpoint", writer);
 }
 
+// Utility function to format libuv pipe information.
+static void ReportPipeEndpoints(uv_handle_t* h, JSONWriter* writer) {
+  uv_any_handle* handle = reinterpret_cast<uv_any_handle*>(h);
+  MallocedBuffer<char> buffer(0);
+  size_t buffer_size = 0;
+  int rc = -1;
+
+  // First call to get required buffer size.
+  rc = uv_pipe_getsockname(&handle->pipe, buffer.data, &buffer_size);
+  if (rc == UV_ENOBUFS) {
+    buffer = MallocedBuffer<char>(buffer_size);
```

Comment:
I think you switched to `MallocedBuffer` because its allocations can fail in a way that can be handled, which is okay in principle but you do actually need to add checks that test whether the allocation was successful in that case, otherwise this will just crash with a segfault when the allocation fails.

(You could probably also just use stack-allocated buffers and truncate the results when `UV_ENOBUFS` occurs, these path lengths are not going to exceed more than, say, 512 bytes, in practice, and that should be okay here)

Reply:
I've updated with checks on MallocedBuffer, PTAL. I'm still wondering what do you mean by truncating the results when UV_ENOBUFS occurs. The uv API will not assign anything to the targeting buffer if the buffer is smaller than expected. There will be no chance to truncate the result.

Reply:
Right, you would have to manually specify a smaller buffer in that case.